src/main/java/org/zalando/baigan/proxy/handler/ContextAwareConfigurationMethodInvocationHandler.java

```diff
@@ -95,6 +106,16 @@ private Object getConfig(final String key) {
             context.put(param, provider.getContextParam(param));
         }
 
+        if (!CollectionUtils.isEmpty(contextProviders)) {
```

Comment:
Yes, it will be a breaking change. However, based on simple github search, we've found that the usage of the context provider by our clients has been minimal due to its less-than-obvious implementation. Instead, clients are opting to add additional logic after fetching configuration values, a task that could have been easily handled by context providers.